### PR TITLE
feat: adding fantom chainlink aggregators to advanced-logic > conversion-supported-currencies

### PR DESCRIPTION
**Pull request by KolevDarko**
Updated the chainlink aggregators for fantom previously but only in currency package, now adding them also in advanced-logic package.
Also updated the aggregators hashes in currency package to be all lowercase, I noticed the other ones are lowercase.

Update: Implemented the suggested change requests in the following way.
Now advanced-logic package gets the chainlink oracles mapping from currency package and then creates it's own mapping that it uses for any-to-eth and any-to-erc20. 

It uses the same logic from the toolbox package, except that toolbox gets it from the contract directly, but advanced-logic here uses the mapping from currency package as mentioned in previous paragraph.

### CODE REVIEW

**Comment by KolevDarko** [comment]
Actually, I need to add fantom tests as well to this

**Review by benjlevesque — APPROVED** [review]

**Review by AdamDowson — APPROVED** [review]

File: packages/currency/src/chainlink-path-aggregators.ts [hunk 1]
```diff
@@ -272,3 +273,71 @@ export function getPath(
   // Get the path
   return route.path(currencyFrom.hash.toLowerCase(), currencyTo.hash.toLowerCase());
 }
+
+/**
+ * These currencies are supported by Chainlink for conversion.
+ */
+export const supportedCurrencies: Record<string, Record<RequestLogicTypes.CURRENCY, string[]>> = {
```
Comment (yomarion) [inline packages/currency/src/chainlink-path-aggregators.ts:280]:
```suggestion
export const currenciesWithConversionOracles: Record<string, Record<RequestLogicTypes.CURRENCY, string[]>> = {
```

**Review by yomarion — COMMENTED** [review]
Are you sure it's a good idea to migrate the `supportedCurrencies` map to the `currency` package ? I have a big doubt, it looks related to one payment network only, so very specific to the advanced logic or payment processor.

**Review by benjlevesque — APPROVED** [review]

File: packages/currency/src/chainlink-path-aggregators.ts (hunk 1 above)
Comment (KolevDarko) [reply packages/currency/src/chainlink-path-aggregators.ts]: I see what you mean. In the meantime I managed to generate the supportedCurrencies map out of the aggregators map in currency package, using the logic from toolbox.
So I will leave the mapping in advanced-logic but generated it out of the aggregator list in /currency package.
The whole idea behind this small refactor was to stop keeping duplicates and have one source of truth.

File: packages/advanced-logic/src/extensions/payment-network/conversion-supported-currencies.ts [hunk 2]
```diff
@@ -1,56 +1,69 @@
...
+    record[type].push(address);
+  }
+  record[type] = record[type].sort();
 };
+
+const generateCurrenciesWithOraclesFromAggregators = (allChainlinkPairs: Record<string, Record<string, Record<string, number>>>):
+  Record<string, Record<RequestLogicTypes.CURRENCY, string[]>> => {
+  const supportedCurrencies: Record<string, Record<RequestLogicTypes.CURRENCY, string[]>> = {};
+  const supportedNetworks = Object.keys(chainlinkCurrencyPairs);
+  // const supportedNetworks = ['private', 'rinkeby', 'mainnet', 'matic', 'fantom'];
```
Comment (yomarion) [inline packages/advanced-logic/src/extensions/payment-network/conversion-supported-currencies.ts:48]: Commented code here FYI

File: packages/toolbox/src/chainlinkConversionPathTools.ts [hunk 3]
```diff
@@ -154,7 +154,7 @@ const addSupportedCurrency = (
   record[type] = record[type].sort();
 };
 
-export const listAggregators = async (options?: IOptions): Promise<void> => {
+export const listAggregators = async (options?: IOptions): Promise<any> => {
```
Comment (yomarion) [inline packages/toolbox/src/chainlinkConversionPathTools.ts:157]: Ideally, can you type the return promise ?

**Review by yomarion — APPROVED** [review]

File: packages/toolbox/src/chainlinkConversionPathTools.ts (hunk 3 above)
Comment (KolevDarko) [reply packages/toolbox/src/chainlinkConversionPathTools.ts:157]: I didn't even end up using this, so I'm returning it to void

**Pull request merged**